graphql_compiler/schema_transformation/merge_schemas.py

```diff
@@ -333,3 +343,373 @@ def _process_generic_type_definition(generic_type, schema_id, existing_scalars,
         )
     merged_schema_ast.definitions.append(generic_type)
     type_name_to_schema_id[type_name] = schema_id
+
+
+def _add_cross_schema_edges(schema_ast, type_name_to_schema_id, scalars, cross_schema_edges,
+                            type_equivalence_hints, query_type):
+    """Add cross schema edges into the schema AST.
+
+    Each cross schema edge will be incorporated into the schema by adding vertex fields
+    with a @stitch directive to relevant vertex types. New fields will be named out_
+    or in_ concatenated with the edge name.
+
+    The type of the new field will either be the type of the opposing vertex specified in
+    the cross schema edge, or the equivalent union type of the type of the opposing vertex
+    if such a union type is specified by type_equivalence_hints.
+
+    New vertex fields will be added to not only each vertex specified by the cross schema
+    edge, but to all of their subclass vertices as well.
+
+    For examples demonstrating the above behaviors, see tests in test_merge_schemas.py that
+    involve subclasses.
+
+    Args:
+        schema_ast: Document. It is modified by this function
```

Comment:
Some more explanation would be helpful here, because I think there are some (currently implicit) expectations of this schema AST -- it's not "just any" schema AST.